timesketch/lib/datastores/elastic.py

```diff
@@ -0,0 +1,173 @@
+# limitations under the License.
+"""ElasticSearch datastore."""
+
+import logging
+
+from elasticsearch import Elasticsearch
+from elasticsearch.exceptions import NotFoundError
+from flask import abort
+
+from timesketch.lib import datastore
+
+# Setup logging
+es_logger = logging.getLogger('elasticsearch')
+es_logger.addHandler(logging.NullHandler())
+
+
+class ElasticSearchDataStore(datastore.DataStore):
+    """Implements the datastore."""
+    def __init__(self, host='127.0.0.1', port=9200):
+        """Create a Elasticsearch client."""
+        self.client = Elasticsearch([
+            {'host': host, 'port': port}
+        ])
+
+    def search(self, sketch_id, query, query_filter, indices):
+        """Search ElasticSearch. This will take a query string from the UI
+        together with a filter definition. Based on this it will execute the
+        search request on ElasticSearch and get result back.
+
+        Args:
+            sketch_id: Integer of sketch primary key
+            query: Query string
+            filters: Dictionary containing filters to apply
+
+        Returns:
+            Set of event documents in JSON format
+        """
+
+        query_dict = {
+            "query": {
+                "filtered": {
+                    "query": {
+                        "query_string": {
+                            "query": query
+                        }
+                    }
+                }
+            },
+            "sort": {
+                "datetime": "asc"
+            }
+        }
+
+        if query_filter.get("star", None):
+            del query_dict['query']['filtered']['query']
+            query_dict['query']['filtered']['filter'] = {
+                "nested": {
+                    "path": "timesketch_label",
+                    "filter": {
+                        "bool": {
+                            "must": [
+                                {
+                                    "term": {
+                                        "timesketch_label.name": "__ts_star"
+                                    }
+                                },
+                                {
+                                    "term": {
+                                        "timesketch_label.sketch_id": sketch_id
+                                    }
+                                }
+                            ]
+                        }
+                    }
+                }
+            }
+
+        if query_filter.get("time_start", None):
+            query_dict['query']['filtered']['filter'] = {
+                "range": {
+                    "datetime": {
+                        "gte": query_filter['time_start'],
+                        "lte": query_filter['time_end']
+                    }
+                }
+            }
+
+        if not indices:
+            return {u'hits':  {u'hits': [], u'total': 0}, u'took': 0}
+
+        # pylint: disable=unexpected-keyword-arg
```

Comment:
Because of how elasticsearch-py adds parameters o the function (with a decorator) which makes pylint sad.